File: tourde.ui/src/models/person.ts
import { Serializable } from "@util/iserializable";

export class Person implements Serializable<Person> {
  constructor(
    public id: number,
    public firstName: string,
    public lastName: string,
    public email: string,
    public phone?: string,
    public dateOfBirth?: string
  ) {}

  deserialize(input: any): Person {
    this.id = input.id;
    this.firstName = input.firstName;
    this.lastName = input.lastName;
    this.email = input.email;
    this.phone = input.phone;
    this.dateOfBirth = input.dateOfBirth;
    return this;
  }
}

File: TourDe.Models/Person.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace TourDe.Models;

[Index(nameof(Email), IsUnique=true)]
public class Person
{ 
    [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public int Id { get; set; }

    [Required, StringLength(50, MinimumLength = 3)]
    public string FirstName { get; set; }

    [Required, StringLength(50, MinimumLength = 3)]
    public string LastName { get; set; }
    
    public DateTime? DateOfBirth { get; set; }

    [Required, EmailAddress]
    public string Email { get; set; }
    
    public string? Phone { get; set; }
}